internal/utils/validation/mgmt.go

```diff
@@ -49,7 +49,7 @@ func GetIncompatibleContracts(ctx context.Context, cl client.Client, release *kc
 	}
 
 	incompatibleContracts := strings.Builder{}
-	for _, p := range mgmt.Spec.Providers {
+	for _, p := range mgmt.Status.RequestedProviders {
```

Comment:
that won't work in case of the disabled validation: that status might (and won't) contain the providers, resulting in successful validation of potentially erroneous configuration